spec/factories/comments.rb

```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :comment, class: Comment do
+    association :commentable, factory: :company
```

Comment:
ok, but company belongs to user, and another user will be created

just don't create user in "after" block. User from comment.company.user